### PR TITLE
add new absent terms

### PR DESCRIPTION
**Pull request by nicolevasilevsky**
addresses #331

### CODE REVIEW

File: src/ontology/pato-edit.obo
Comment (dosumis) [inline src/ontology/pato-edit.obo:20603]: an -> a

File: src/ontology/pato-edit.obo
Comment (dosumis) [inline src/ontology/pato-edit.obo:20594]: an -> a

**Comment by dosumis** [comment]
@balhoff  - can you take a look at this?  I'd like to add you as a reviewer, but you may need to accept editor's rights first.

File: src/ontology/pato-edit.obo
Comment (dosumis) [inline src/ontology/pato-edit.obo:20593]: Sounds a bit too much like it refers to a physical entity rather than a quality.  Maybe 'absence of physical object' ?

File: src/ontology/pato-edit.obo
Comment (dosumis) [inline src/ontology/pato-edit.obo:20612]: Sounds a bit too much like it refers to a physical entity rather than a quality. Maybe 'absence of anatomical entity' ?

**Comment by sbello** [comment]
I've added Yvonne to the review list as well.

**Comment by balhoff** [comment]
What are these qualities of? Is this using the same EQ model as some other quality? `pointed and inheres_in some ear` vs. `'absent anatomical entity' and inheres_in some ear`? Or did you come up with another model?

**Comment by dosumis** [comment]
@balhoff - I'm afraid it's just the same old problematic patterns, extended to ensure that absent anatomical entity phenotypes end up under abnormal morphology phenotypes for that entity and - which fits with editor expectations.

TBH - I still prefer your solution here: https://arxiv.org/pdf/1410.3862.pdf (I strongly recommend that everyone following this pull request ant the associated ticket read it).  It's the only one that is comprehensive and consistent (in the general sense as well as the logical sense).  But we're still stuck trying to cope with irreconcilable expectations about classification and term interpretation.  The main clash is:

  * **A**.  Expectation that absent X phenotypes should classify under general X phenotypes e.g. absent limb classifies under abnormal limb (at a minimum) but ideally under abnormal limb morphology, perhaps also under reduced limb size (limit point of size reduction is absence).   

***VS***

* **B**.   Expectation that existential assertions about some entity imply its presence (basic logic) - something your paper solves elegantly

I don't think there's any way to reconcile these.

**A** requires 'absent X' to be interpreted as something like  'absence of some (type of ) X"   That way inferences like this make sense:

absent tooth (the singular helps interpretation here here)
.    < - SubClassOf absent canine tooth

BUT...  This doesn't fit editor expectations either.  Here's MP (MGI browser - no inference - showing intention)

image

Here's the same (with inference - OLS):

image

Using punning and value restriction would (as in https://arxiv.org/pdf/1410.3862.pdf) would fix this, but also lose the inferred placement under abnormal tooth morphology.  We need this inference (& similar) for classify species neutral classes in uPheno2 and Phenotypes in XPO, both of which have no asserted manual classification (see XPO example here https://github.com/obophenotype/xenopus-phenotype-ontology/issues/125)

**Comment by matentzn** [comment]
One way around this may be to use non logical modelling, like skos:broadMatch or something along these lines to cover "expectations about classification" and help groups that use it to implement it into their hierarchical browsers..

**Comment by dosumis** [comment]
One possible compromise:

Define 'abnormal x phenotype' as disjointUnionOf: 'abnormal x morphology' and 'absent x'.  **_We'd have to give up on nesting absence under morphology_**, but we would have grouping classes at the top level - with abnormal X phenotype terms.  This works nicely with HermiT, but DisjointUnion of is outside of EL.  However, I think we could modularise to infer (or even have some bespoke assertion job) =>  classification immediately under these top level terms

**Comment by dosumis** [comment]
We can still get terms into the morphology branch, just under the morphology for parent structure in the partonomy.

absent femur: `'lacks all parts of type' and (inheres_in some hindlimb) and (towards value femur)`

`'lacks all parts of type' SubPropertyOf morphology`

hindlimb morphology phenotype: `morphology and inheres_in some hindlimb`

=> 
image

File: src/ontology/pato-edit.obo
Comment (nicolevasilevsky) [reply src/ontology/pato-edit.obo]: updated

File: src/ontology/pato-edit.obo
Comment (nicolevasilevsky) [reply src/ontology/pato-edit.obo]: updated

File: src/ontology/pato-edit.obo
Comment (nicolevasilevsky) [reply src/ontology/pato-edit.obo]: I renamed this

File: src/ontology/pato-edit.obo
Comment (nicolevasilevsky) [reply src/ontology/pato-edit.obo]: I renamed this

**Comment by nicolevasilevsky** [comment]
@dosumis @matentzn @balhoff is there agreement on the approach needed here? Should we close this PR or are there steps to move this forward?
thanks!

**Comment by sbello** [comment]
I'll add this to the agenda for the next UPheno call. I think the changes look good but I was at the meeting. I think the group should review and after the call we can decide if we want to stick with this approach.

**Comment by nicolevasilevsky** [comment]
great, thanks @shawntanzk! I'll hold off on fixing the conflict until it is confirmed that we want to move forward.

**Comment by sbello** [comment]
Discussed on the UPheno call 7/28
suggest that absence of anatomical entity be a child of absence of physical object as well as a child of morphology

**Review by sbello — CHANGES_REQUESTED** [review]
see earlier comment for suggested changes from 7/28/22 call

**Comment by nicolevasilevsky** [comment]
Thanks @sbello, I'll work on this :)

File: src/ontology/pato-edit.obo
Comment (nicolevasilevsky) [inline src/ontology/pato-edit.obo:20820]:
```suggestion
id: PATO:0040058
```

File: src/ontology/pato-edit.obo
Comment (nicolevasilevsky) [inline src/ontology/pato-edit.obo:20829]:
```suggestion
id: PATO:0040059
```

File: src/ontology/pato-edit.obo
Comment (nicolevasilevsky) [inline src/ontology/pato-edit.obo:20839]:
```suggestion
id: PATO:0040060
```

File: src/ontology/pato-edit.obo
Comment (nicolevasilevsky) [inline src/ontology/pato-edit.obo:20845]:
```suggestion
is_a PATO:0040058 ! absence of physical object
created_by: http://orcid.org/0000-0001-5208-3432
```

**Comment by nicolevasilevsky** [comment]
- assigned new ID to absence of anatomical entity b/c another term is now using the former ID (new ID = PATO:0040060)
- assigned new ID to absence of physical object  b/c another term is now using the former ID (new ID = PATO:0040058)
- assigned new ID to absent process  b/c another term is now using the former ID (new ID = PATO:0040059)
- added parent to absence of anatomical entity: absence of physical object

**Comment by nicolevasilevsky** [comment]
1. @sbello should I announce that we will obsolete  ‘absent’ and will switch to using the new terms?
1. Could you create a new ticket for the absent some vs absent all issue?
1. I'll follow up with Chris Grove about updating the patterns after this is merged
1. I'll work with Nico on fixing the imports after this is merge

**Comment by nicolevasilevsky** [comment]
Lastly, @sbello thanks for following up on this issue on the call! 😸

**Pull request merged**
